libobs/util/cf-lexer.c

```diff
@@ -71,22 +71,26 @@ char *cf_literal_to_str(const char *literal, size_t count)
 
 	if (count < 2)
 		return NULL;
-	if (literal[0] != literal[count-1])
+	if (literal[0] != literal[count - 1])
 		return NULL;
 	if (literal[0] != '\"' && literal[0] != '\'')
 		return NULL;
-
-	str = bmalloc(count - 1);
-	temp_src = literal;
+	//bzalloc so that the string is memset w/ 0s
+	str = bzalloc(count - 1);
+	//strip leading character
+	temp_src = literal + 1;
 	temp_dst = str;
-
-	while (*temp_src) {
+	//strip trailing character
+	count -= 2;
+	while (*temp_src && count > 0) {
 		if (*temp_src == '\\') {
 			temp_src++;
 			cf_convert_from_escape_literal(&temp_dst, &temp_src);
-		} else {
+		}
+		else {
```

Comment:
You're changing code styling here unnecessarily.  Additionally, for this project it should always be

```C
} else {
```